services/github/github-milestone-detail.service.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { metric } = require('../text-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { GithubAuthV3Service } = require('./github-auth-service')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+
+const schema = Joi.object({
+  open_issues: nonNegativeInteger,
+  closed_issues: nonNegativeInteger,
+  title: Joi.string(),
+}).required()
+
+module.exports = class GithubMilestoneDetail extends GithubAuthV3Service {
+  static get category() {
+    return 'issue-tracking'
+  }
+
+  static get route() {
+    return {
+      base: 'github/milestones',
+      pattern:
+        ':variant(issues-closed|issues-open|issues-total|progress|progress-percent)/:user/:repo/:number([0-9]+)',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub milestone',
+        namedParams: {
+          variant: 'issues-open',
+          user: 'badges',
+          repo: 'shields',
+          number: '1',
+        },
+        staticPreview: {
+          label: 'milestone issues',
+          message: '17 / 22',
+          color: 'blue',
+        },
+        documentation,
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'milestones',
+      color: 'informational',
+    }
+  }
+
+  static render({ user, repo, variant, number, milestone }) {
+    let milestoneMetric
+    let color
+    let label = ''
+
+    switch (variant) {
+      case 'issues-open':
+        milestoneMetric = milestone.open_issues
+        color = 'red'
+        label = 'open issues'
+        break
+      case 'issues-closed':
+        milestoneMetric = milestone.closed_issues
+        color = 'green'
+        label = 'closed issues'
+        break
+      case 'issues-total':
+        milestoneMetric = milestone.open_issues + milestone.closed_issues
+        color = 'blue'
+        label = 'issues'
+        break
+      case 'progress':
+        milestoneMetric = `${
+          milestone.closed_issues
+        } / ${milestone.open_issues + milestone.closed_issues}`
```

Comment:
There are a few other places where we use `/` in the badge's text, but we tend not to put surrounding spaces (see examples in the downloads or rating categories).